### PR TITLE
Endpoints

### PR DESCRIPTION
**Pull request by EricLarson2020**
Created endpoints for most recent and advanced search functionality

### CODE REVIEW

File: app/poros/nytimes_results.rb [hunk 1]
```diff
@@ -30,6 +30,23 @@ def get_obituaries_by_date(beginning_date, ending_date)
...
   end
 
+  def get_most_recent_obituaries
+    json = NytimesService.new.get_most_recent_obituaries
+    json[:response][:docs].map do |obituary_data|
+      Obituary.new(format_obituary_data(obituary_data))
+    end
+  end
+
+  def get_advanced_obituaries(name, date)
```
Comment (EricLarson2020) [inline app/poros/nytimes_results.rb:45]: Using the name and date as params here, I will have it ignore the state, age, and city params if they use the NYT API

File: app/test/obituary_test.rb [hunk 2]
```diff
@@ -33,4 +33,14 @@ def test_it_returns_formatted_obituary_data_by_date
 			get '/date', beginning_date: '20170101', ending_date: '20190101'
 			assert_equal "obituary", JSON.parse(last_response.body).first['data']['type']
 	end
+
+	def test_returns_most_recent_obituaries
```
Comment (EricLarson2020) [inline app/test/obituary_test.rb:38]: Will build out these test to better reflect what is returned, as discussed in meeting, on another branch

Comment (jpc20) [reply app/test/obituary_test.rb:38]: Sounds good, yeah it definitely makes sense to start testing edge cases and sad paths here.

**Review by jpc20 — COMMENTED** [review]
This all looks good Eric.
Have you been able to start debugging the JSON parser errors we are randomly getting? I have been following the breadcrumbs and I have been able to get back to the `NytimesResults` before the error hits. It seems like it might be an error coming from our request to the NYTimes API, but I am still not sure.

File: app/poros/nytimes_results.rb [hunk 3]
```diff
@@ -30,6 +30,23 @@ def get_obituaries_by_date(beginning_date, ending_date)
...
+
+    json = NytimesService.new.get_advanced_obituaries(name, date)
+    json[:response][:docs].map do |obituary_data|
+      Obituary.new(format_obituary_data(obituary_data))
```
Comment (jpc20) [inline app/poros/nytimes_results.rb:49]: Nice job separating these into their own methods.

**Review by jpc20 — APPROVED** [review]
Nice work, Eric. I think we agreed the error we are getting is that we are hitting the 10 requests per minute limit, which shouldn't be a big deal.

File: app/poros/nytimes_results.rb (hunk 1 above)
Comment (stellakunzang) [reply app/poros/nytimes_results.rb:45]: That makes sense. I guess we will have to figure out a way to let users know on the advanced search page that state, age and city will only search our community obits.

File: app/services/nytimes_service.rb [hunk 4]
```diff
@@ -38,6 +38,22 @@ def get_obituaries_by_date(beginning_date, ending_date)
...
+      req.params['sort'] = 'newest'
+    end
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def get_advanced_obituaries(name, year)
+
+    response = conn.get do |req|
+      req.params['fq'] = "section_name:(obituaries) AND headline:(#{name}) AND pub_year:(#{year})"
+    end
```
Comment (stellakunzang) [inline app/services/nytimes_service.rb:53]: yay this is so awesome!

**Review by stellakunzang — APPROVED** [review]
Thank you Eric, this looks amazing!

File: app/poros/obituary.rb [hunk 5]
```diff
@@ -3,12 +3,27 @@ class Obituary
...
+
+  def sort_images(data)
+
+    if data.empty?
+      nil
+    else
+      data.select do |entry|
+        entry[:legacy][:thumbnail] != nil
+      end.first[:legacy][:thumbnail]
+    end
```
Comment (stellakunzang) [inline app/poros/obituary.rb:27]: This is so brilliant and I am so grateful that you took the time to write this simple, elegant code which accomplishes something that would have taken me twice as many lines and been really janky!

**Review by stellakunzang — APPROVED** [review]
This microservice is amazing and I'm so excited about the code you have written. It is so clean and readable and efficient, and I super appreciate your willingness to go back in and write more thorough tests.

File: app/poros/obituary.rb [hunk 6]
```diff
@@ -3,12 +3,27 @@ class Obituary
...
+              :image
+
   def initialize(obituary_data)
+
     @headline = obituary_data[:headline]
     @abstract = obituary_data[:abstract]
     @web_url = obituary_data[:web_url]
     @pub_date = obituary_data[:pub_date]
     @word_count = obituary_data[:word_count]
+    @image = sort_images(obituary_data[:image])
```
Comment (EricLarson2020) [inline app/poros/obituary.rb:16]: If no thumbnail present returns nil, else selects thumbnail

File: app/poros/obituary.rb (hunk 5 above)
Comment (jpc20) [reply app/poros/obituary.rb:27]: Yes, this looks great Eric!

File: app/poros/nytimes_results.rb (hunk 1 above)
Comment (sagemlee) [reply app/poros/nytimes_results.rb:45]: I imaged have a little line between the search fields to have a slight sectioning off.

File: app/poros/obituary.rb (hunk 6 above)
Comment (sagemlee) [reply app/poros/obituary.rb:16]: Awesome!

File: app/services/nytimes_service.rb (hunk 4 above)
Comment (sagemlee) [reply app/services/nytimes_service.rb:53]: Noooicee

**Review by sagemlee — APPROVED** [review]
